crates/sui-verifier/src/entry_points_verifier.rs

```diff
@@ -252,18 +254,26 @@ fn is_primitive(
     }
 }
 
-pub fn is_tx_context(view: &BinaryIndexedView, p: &SignatureToken) -> bool {
+// Returns Some(is_mut) if the type is a reference to the TxnContext. is_mut being true with
+// a MutableReference, and false otherwise.
+// Returns None for all other types
+pub fn is_tx_context(view: &BinaryIndexedView, p: &SignatureToken) -> Option<bool> {
```

Comment:
I think an `enum` or newtype would be clearer here --I keep incorrectly thinking the `tx_context_mut.is_some()`  check elsewhere is true only when you get a mutable txcontext.

Reply:
It's maybe a bit obtuse not using an `Option`, but at least we have to manually check it each time, and I prefer that over hiding cases behind methods